src/main/java/com/minecolonies/coremod/commands/killcommands/AbstractKillCommand.java

```diff
@@ -0,0 +1,86 @@
+package com.minecolonies.coremod.commands.killcommands;
+
+import com.minecolonies.coremod.commands.AbstractSingleCommand;
+import com.minecolonies.coremod.commands.DeleteCommand;
+import com.minecolonies.coremod.commands.MinecoloniesCommand;
+import net.minecraft.command.CommandException;
+import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TextComponentString;
+import org.jetbrains.annotations.NotNull;
+
+import javax.annotation.Nullable;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Abstract command for killing all entities of Type T on map.
+ */
+public abstract class AbstractKillCommand<T extends Entity> extends AbstractSingleCommand
+{
+
+    /**
+     * Gets the DESC string of the command (Entity name usually).
+     *
+     * @return The {@link String} of the Desc.
+     */
+    public abstract String getDesc();
+
+    /**
+     * Gets the class of the entity we're killing.
+     *
+     * @return The {@link Class} of the Entity.
+     */
+    abstract Class<T> getEntityClass();
+
+    /**
+     * Initialize this SubCommand with it's parents.
+     *
+     * @param desc the command name.
+     */
+    public AbstractKillCommand(@NotNull final String desc)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this constructor to comply with Java Code Conventions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1213)